modules/tax/src/Plugin/Commerce/TaxType/CanadianSalesTax.php

```diff
@@ -0,0 +1,261 @@
+          ]
+        ]
+      ]
+    ]);
+
+    $zones['nb'] = new TaxZone([
+      'id' => 'nb',
+      'label' => $this->t('New Brunswick Harmonized Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'NB']
+      ],
+      'rates' => [
+        [
+          'id' => 'HST',
+          'label' => $this->t('HST'),
+          'amounts' => [
+            ['amount' => '0.15', 'start_date' => '2016-07-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    $zones['nl'] = new TaxZone([
+      'id' => 'nl',
+      'label' => $this->t('Newfoundland Harmonized Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'NL']
+      ],
+      'rates' => [
+        [
+          'id' => 'HST',
+          'label' => $this->t('HST'),
+          'amounts' => [
+            ['amount' => '0.15', 'start_date' => '2016-07-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    $zones['ns'] = new TaxZone([
+      'id' => 'ns',
+      'label' => $this->t('Nova Scotia Harmonized Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'NS']
+      ],
+      'rates' => [
+        [
+          'id' => 'HST',
+          'label' => $this->t('HST'),
+          'amounts' => [
+            ['amount' => '0.15', 'start_date' => '2010-07-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    $zones['on'] = new TaxZone([
+      'id' => 'on',
+      'label' => $this->t('Ontario Harmonized Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'ON']
+      ],
+      'rates' => [
+        [
+          'id' => 'HST',
+          'label' => $this->t('HST'),
+          'amounts' => [
+            ['amount' => '0.13', 'start_date' => '2010-07-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    $zones['pe'] = new TaxZone([
+      'id' => 'pe',
+      'label' => $this->t('PEI Harmonized Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'PE']
+      ],
+      'rates' => [
+        [
+          'id' => 'HST',
+          'label' => $this->t('HST'),
+          'amounts' => [
+            ['amount' => '0.14', 'start_date' => '2013-04-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    $zones['qc'] = new TaxZone([
+      'id' => 'qc',
+      'label' => $this->t('Quebec Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'QC']
+      ],
+      'rates' => [
+        [
+          'id' => 'QST',
+          'label' => $this->t('QST'),
+          'amounts' => [
+            ['amount' => '0.09975', 'start_date' => '2013-01-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    return $zones;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function resolveZones(OrderItemInterface $order_item, ProfileInterface $customer_profile) {
+    $zones = $this->getZones();
+    $customer_address = $customer_profile->address->first();
```

Comment:
We should do a quick check of $customer_address->getCountryCode() == 'CA' before proceeding with the zone match.